src/main/kotlin/com/github/mitchellolsthoorn/testgenie/settings/TestGenieSettingsComponent.kt

```diff
@@ -12,34 +14,167 @@ import javax.swing.JTextField
+    private var criterionLineCheckBox = JCheckBox("Line coverage")
+    private var criterionBranchCheckBox = JCheckBox("Branch coverage")
+    private var criterionExceptionCheckBox = JCheckBox("Exception coverage")
+    private var criterionWeakMutationCheckBox = JCheckBox("Mutation coverage")
+    private var criterionOutputCheckBox = JCheckBox("Output coverage")
+    private var criterionMethodCheckBox = JCheckBox("Method coverage")
+    private var criterionMethodNoExceptionCheckBox = JCheckBox("Method no exception coverage")
+    private var criterionCBranchCheckBox = JCheckBox("CBranch coverage")
+    private var minimizeCheckBox = JCheckBox("Minimize test suite after generation")
 
     init {
         panel = FormBuilder.createFormBuilder()
-            .addLabeledComponent(JBLabel("Global timeout "), globalTimeoutTextField, 1, false)
-            .addComponent(showCoverageCheckBox, 1)
+            .addComponent(JXTitledSeparator("General settings"))
+            .addComponent(showCoverageCheckBox, 10)
+            .addComponent(sandboxCheckBox, 10)
+            .addComponent(assertionsCheckBox, 10)
+            .addLabeledComponent(JBLabel("Seed(random if left empty) "), seedTextField, 10, false)
+            .addLabeledComponent(JBLabel("Select search algorithm"), algorithmSelector, 10, false)
+            .addLabeledComponent(JBLabel("Select configuration id (null if left empty) "), configurationIdTextField, 5, false)
+            .addComponent(clientOnThreadCheckBox, 10)
+            .addComponent(minimizeCheckBox, 10)
+            .addComponent(junitCheckCheckBox, 10)
+            .addComponent(criterionSeparator, 15)
+            .addComponent(criterionLineCheckBox, 5)
+            .addComponent(criterionBranchCheckBox, 5)
+            .addComponent(criterionExceptionCheckBox, 5)
+            .addComponent(criterionWeakMutationCheckBox, 5)
+            .addComponent(criterionOutputCheckBox, 5)
+            .addComponent(criterionMethodCheckBox, 5)
+            .addComponent(criterionMethodNoExceptionCheckBox, 5)
+            .addComponent(criterionCBranchCheckBox, 5)
             .addComponentFillVertically(JPanel(), 0)
             .panel
+
+        algorithmSelector.setMinimumAndPreferredWidth(300)
+        configurationIdTextField.toolTipText = "Label that identifies the used configuration of EvoSuite. This is only done when running experiments."
+        clientOnThreadCheckBox.toolTipText = "Run client process on same JVM of master in separate thread. To be used only for debugging purposes"
+        junitCheckCheckBox.toolTipText = "Compile and run resulting JUnit test suite (if any was created)"
+        criterionSeparator.toolTipText = "Coverage criterion. Can define more than one criterion by checking multiple checkboxes. " +
+                "\n By default, all are used."
     }
 
     /**
      * Returns the UI component that should be focused when a user opens the TestGenie Settings page.
```

Comment:
How about `@return` in KDoc? Or it will be addressed in a separate issue?

Reply:
thanks, will add.